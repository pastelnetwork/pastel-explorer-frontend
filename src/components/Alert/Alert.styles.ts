import styled from 'styled-components/macro';

import AlertTitle from '@material-ui/lab/AlertTitle';

export const Title = styled(AlertTitle)`
  font-weight: bold;
`;
