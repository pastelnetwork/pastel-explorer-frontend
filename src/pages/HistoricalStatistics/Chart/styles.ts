import { makeStyles } from '@material-ui/core/styles';
import { TAppTheme } from '@theme/index';

export const eChartLineStyles = makeStyles((theme: TAppTheme) => ({
  container: {
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    height: '100%',
    width: '100%',
    padding: '0 20px',
    textAlign: 'center',
    boxSizing: 'border-box',
    position: 'relative',
    color: 'rgb(171, 170, 193)',
    [theme.breakpoints.down('sm')]: {
      padding: 0,
    },
  },
  lineChartWrap: {
    width: '100%',
    height: 'auto',
    marginBottom: 20,
  },
  lineChartContainer: {
    display: 'block',
    width: '100%',
    height: '100%',
    backgroundColor: '#212121',
  },
  lineChartTitle: {
    width: '100%',
    textAlign: 'left',
    fontSize: '30px',
    [theme.breakpoints.down('sm')]: {
      fontSize: 15,
    },
  },
  lineChartHeader: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
    paddingTop: '30px',
    paddingBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      paddingTop: 10,
      paddingLeft: 10,
      justifyContent: 'flex-start',
      flexDirection: 'column',
    },
  },
  uploadButton: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: '#6151d0',
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '14px',
    padding: '8px 16px',
    borderRadius: '4px',
    marginLeft: '8px',
    marginRight: '8px',
    '&:hover': {
      cursor: 'pointer',
      opacity: 0.8,
    },
  },
  reactECharts: {
    width: '100%',
    height: 'calc(100vh - 450px) !important',
  },
  granularitySelect: {
    display: 'flex',
    width: '100%',
    maxWidth: 200,
    justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: 15,
    marginRight: 50,
  },
  periodSelect: {
    display: 'flex',
    minWidth: '200px',
    // justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: '15px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 5,
    },
  },
  lineChartDownloadButtonBar: {
    display: 'flex',
    width: '100%',
    justifyContent: 'flex-end',
    height: '30px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 10,
      justifyContent: 'flex-start',
    },
  },
  lineChartFooter: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-around',
    alignItems: 'center',
    paddingBottom: 30,
    [theme.breakpoints.down('sm')]: {
      flexWrap: 'wrap',
      justifyContent: 'flex-start',
      paddingBottom: 10,
    },
  },
  lineChartThemeSelect: {
    display: 'flex',
    minWidth: '150px',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  themeSelectButton: {
    width: '30px',
    height: '30px',
    borderRadius: '5px',
  },
  filterButton: {
    backgroundColor: '#eee',
    color: '#212124',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '12px',
    padding: '4px 8px',
    borderRadius: '3px',
    borderWidth: '1px',
    borderColor: '#212124',
    marginLeft: '4px',
    marginRight: '4px',
    '&:hover': {
      cursor: 'pointer',
      opacity: '0.8',
    },
  },
  activeButton: {
    backgroundColor: '#c3921f',
    borderColor: '#c3921f',
    color: '#fff',
  },
  activeThemeButton: {
    borderColor: '#008000',
    borderWidth: '5px',
  },
}));
