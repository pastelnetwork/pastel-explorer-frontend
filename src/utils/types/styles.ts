/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/ban-types */
import { SpacingProps } from '@material-ui/system';
import { Theme } from '@material-ui/core';

export type GlobalStyleProps = {
  theme: Theme & { palette: any };
};

export interface MuiButtonSpacingType extends SpacingProps {
  component?: React.PropsWithoutRef<{}>;
  to?: string;
}
