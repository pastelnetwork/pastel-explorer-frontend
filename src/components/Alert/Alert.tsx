import * as React from 'react';
import Snackbar from '@mui/material/Snackbar';
import Alert, { AlertProps } from '@mui/material/Alert';

import * as Styles from './Alert.styles';

interface AlertComponentProps {
  isOpen: boolean;
  severity?: AlertProps['severity'];
  autoHideDuration?: number;
  message?: React.ReactNode;
  title?: React.ReactNode;
  outsideClickClose?: boolean;
  onClose?: () => void;
}

const AlertComponent: React.FC<AlertComponentProps> = ({
  isOpen,
  onClose,
  severity = 'success',
  autoHideDuration = 4000,
  outsideClickClose = true,
  message = '',
  title = '',
}) => {
  const [isAlertOpen, setIsAlertOpen] = React.useState(isOpen);

  React.useEffect(() => setIsAlertOpen(isOpen), [isOpen]);

  const handleClose = () => {
    setIsAlertOpen(false);
    return onClose && onClose();
  };

  const handleOutsideClick = () => (outsideClickClose ? handleClose() : undefined);

  return (
    <Snackbar open={isAlertOpen} autoHideDuration={autoHideDuration} onClose={handleOutsideClick}>
      <Alert onClose={handleClose} severity={severity}>
        {title && <Styles.Title>{title}</Styles.Title>}
        {message}
      </Alert>
    </Snackbar>
  );
};

export default AlertComponent;

AlertComponent.defaultProps = {
  severity: 'success',
  autoHideDuration: 4000,
  message: '',
  title: '',
  outsideClickClose: true,
  onClose: () => {},
};
