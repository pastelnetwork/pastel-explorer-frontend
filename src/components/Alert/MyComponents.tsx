import Snackbar from '@material-ui/core/Snackbar';
import Alert from '@material-ui/lab/Alert';

import * as Styles from './Alert.styles';

const MyComponent = () => {
  return (
    <Snackbar open>
      <Alert severity="success">
        <button type="button">Click</button>
        <Styles.Title className="my-title-class">Title</Styles.Title>
        Content
      </Alert>
    </Snackbar>
  );
};

export default MyComponent;
