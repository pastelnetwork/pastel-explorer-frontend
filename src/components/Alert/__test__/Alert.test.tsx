import { shallow } from 'enzyme';
import MuiAlert from '@material-ui/lab/Alert';
import Snackbar from '@material-ui/core/Snackbar';

import 'jest-styled-components';

import Alert from '../Alert';
import * as Styles from '../Alert.styles';

describe('components/Alert', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <Alert
        isOpen
        title="Lorem ipsum dolor sit amet"
        message="Ut eu dignissim lacus, a volutpat mauris."
      />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(MuiAlert).length).toEqual(1);
    expect(wrapper.find(Snackbar).length).toEqual(1);
    expect(wrapper.find(Styles.Title).length).toEqual(1);
  });
});
