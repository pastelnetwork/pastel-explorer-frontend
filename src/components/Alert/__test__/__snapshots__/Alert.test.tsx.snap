// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Alert renders correctly 1`] = `ShallowWrapper {}`;
